scss/_settings_colors.scss

```diff
@@ -77,6 +79,7 @@ $colors--dark-theme--text-muted: rgba($colors--dark-theme--text-default, 0.55) !
 $colors--dark-theme--background-default: hsl(0, 0%, 15%) !default;
 $colors--dark-theme--background-alt: hsl(0, 0%, 20%) !default;
 $colors--dark-theme--background-highlight: rgba($colors--dark-theme--text-default, 0.05) !default;
+$colors--dark-theme--background-overlay: transparentize($color-dark, 0.15) !default;
```

Comment:
Should this switched to `$color-light` to add contrast to the dark theme version?

Reply:
@lyubomir-popov ? What should be the color of overlay in dark theme?

Reply:
In the design, the overlay styling was taken from the RBAC overlay, which is 40% #111 for the light theme.  

Inverting to light overlay seems strange, the background has a high contrast:
![image](https://user-images.githubusercontent.com/2741678/79358624-58543b00-7f39-11ea-8c0e-314aee77aadc.png)

Dark p strip underneath not ideal too at this opacity:
![image](https://user-images.githubusercontent.com/2741678/79356582-10ccaf80-7f37-11ea-9528-ff6306ae1c92.png)

I think dark overlay is better for both, and I much prefer the .85 opacity used by @bartaz. In rbac, the user is supposed to be able make out what is underneath while the overlay is on, but here that's not necessary. So I think what is currently happening on this pr is best:
![image](https://user-images.githubusercontent.com/2741678/79357487-0232c800-7f38-11ea-816d-15b72546c47d.png)

Separately, can we have a box-shadow too, like cards? Then once we have  a proposal for container shadows/elevaiton we can go back and update the shadow depth to something more suitable.

Reply:
@lyubomir-popov What would be the purpose of the box shadow if we have an overlay?

Dark box shadow is completely invisible on such overlay.

It is visible when it shouldn't (when drawer is offscreen it sticks from the left):
<img width="137" alt="Screenshot 2020-04-16 at 07 38 18" src="https://user-images.githubusercontent.com/83575/79418657-765f8100-7fb5-11ea-9f65-4558b5b80e85.png">

We can prevent this from happening of course, but why applying shadow if it will be not visible when drawer is open anyway?

Reply:
I did think drop shadow would help but may need to be increased from the normal card shadow. Also, I was thinking it might look nice if we transition the shadow spread. So off canvas has 0 spread were as when the menu is revealed the shadowed spread is set and transitioned to at the same rate as the slide out. This would solve the shadow visible off screen and might make the shadow more visible to the eye.

Reply:
Box shadow will be useful, if it's visible on top of overlay. We would need to adjust the opacity of overlay and shadow to achieve that. With overlay appearing immediately when sidenav starts animating into the screen there is no way to make shadow visible (with our current colors).

Animating shadow is an interesting idea. But I wonder if this will not make it look like drawer is 'elevating' during animation, which could be a bit unexpected.

Reply:
We recently created a set of rules for different things, one of them was anything that overlaps something else should have a box shadow, so it's a consistency thing. I was wary of suggesting a bespoke drop shadow, but ideally it should be something like the google maps one: 
![image](https://user-images.githubusercontent.com/2741678/79434399-39e85100-7fc6-11ea-9e3f-f1b528fb2403.png) we just have to define our own in the shadow hierarchy piece of work.

I know it won't be very visible here, but in the future we may change the opacity of the overlay. It doesn't hurt to apply the rules consistently.